tensorflow_io/python/ops/parquet_dataset_ops.py

```diff
@@ -58,15 +47,28 @@ def dtype_f(dtypes, components, column):
                 dtype = tf.as_dtype(dtype.numpy())
                 return dtype
 
-            if columns is not None:
+            if not tf.executing_eagerly():
+                assert columns is not None
+                assert isinstance(columns, dict)
```

Comment:
@dulacp I think we should raise exceptions with proper messages here instead of an `assert` so that users can better understand the issue.

Reply:
Thanks for pointing it out. I changed this into a `ValueError` exception and added two tests to validate the behavior and avoid regressions.